packages/manager/apps/container/src/payment-modal/PaymentModal.tsx

```diff
@@ -38,14 +38,15 @@ const computeAlert = (paymentMethods: IPaymentMethod[]): string => {
     return PAYMENT_ALERTS.NO_DEFAULT;
   }
   const currentCreditCard: IPaymentMethod = paymentMethods?.find(currentPaymentMethod => currentPaymentMethod.paymentType === 'CREDIT_CARD');
+  
   if (currentCreditCard) {
     const creditCardExpirationDate = new Date(currentCreditCard.expirationDate);
     if (creditCardExpirationDate.getTime() < Date.now()) {
       return PAYMENT_ALERTS.EXPIRED_CARD;
     }
-    const currentDateMinus30Days = new Date();
-    currentDateMinus30Days.setDate(currentDateMinus30Days.getDate() - 30);
-    const isSoonToBeExpireCreditCard = currentDateMinus30Days.getTime() > Date.now() && creditCardExpirationDate.getTime() > Date.now();
+    const currentDatePlus30Days = new Date();
```

Comment:
As seen together, this seems to be working as intended, but when reading the code we can understand that we're taking the current date, adding it 30 days and compared it to the expiration date which feels like the reverse logic.
So either add a comment to explain that it is in the end the same as testing that the current date is less than 30 days before the expiration date, or use the expiration date, substract 30 days on it and compare it to the current date